### ISSUE TITLE
[A11y] ScreenReader doesn't announce ScrollStateChanges in CustomScrollView (e.g "showing items 1 to 10 of 23")

### ISSUE THREAD

**Issue by hoschiho** [flutter/flutter#136246]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Have a CustomScrollView with 'semanticChildCount' set, 'addSemanticIndexes: true' and a bunch of elements
2. Turn VoiceOver / TalkBack on
3. Perform a ScrollGesture (on iOS Three-finger swipe up)

The feature is described in detail here in the section 'Accessibility': https://api.flutter.dev/flutter/widgets/CustomScrollView-class.html 

### Expected results

VoiceOver should announce something like "showing items 1 to 10 of 23"

### Actual results

Accessibility focus moves to the title and announces it.

### Code sample

<details><summary>Code sample</summary>

```dart
CustomScrollView(
        semanticChildCount: 100,
        slivers: <Widget>[
          SliverList(
            delegate: SliverChildBuilderDelegate(
              addSemanticIndexes: true,
              (BuildContext context, int index) {
                // Build your list item here
                return SizedBox(
                  height: 150,
                  child: Text('Item $index'),
                );
              },
              childCount: 100, // Number of items in the list
            ),
          ),
        ],
      ),
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

https://github.com/flutter/flutter/assets/57287592/3b27cf95-e059-4248-a7c0-f8d76b829a16

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[✓] Flutter (Channel stable, 3.13.0, on macOS 13.0 22A8380 darwin-arm64, locale en-CH)
    • Flutter version 3.13.0 on channel stable at /Users/pascalhostettler/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision efbf63d9c6 (8 weeks ago), 2023-08-15 21:05:06 -0500
    • Engine revision 1ac611c64e
    • Dart version 3.1.0
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/pascalhostettler/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.82.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.74.0

[✓] Connected device (6 available)
    • iPhone 14 (mobile)                             • 78EAF084-EFE3-408C-AABA-2AC7707DAEF9     • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • iPad Pro (12.9-inch) (6th generation) (mobile) • 80B2B170-275B-425C-A3A9-22BD0E9910E4     • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-4 (simulator)
    • macOS (desktop)                                • macos                                    • darwin-arm64   • macOS 13.0 22A8380 darwin-arm64
    • Chrome (web)                                   • chrome                                   • web-javascript • Google Chrome 117.0.5938.149

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by huycozy** [comment]
Hi @hoschiho 
There is a known issue https://github.com/flutter/flutter/issues/93232 for a scrolling view with a11y as well. Closing this issue in favor of that one. Please leave this case with `CustomScrollView` there and react an upvote to the original comment will help to raise the issue. Thank you!
